src/wp-admin/includes/image.php

```diff
@@ -291,30 +291,55 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		 * If the original image's dimensions are over the threshold,
 		 * scale the image and use it as the "full" size.
 		 */
+		$scale_down = false;
+		$convert    = false;
+
 		if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
+			// The image will be converted if needed on saving.
+			$scale_down = true;
+		} else {
+			// The image may need to be converted regardless of its dimensions.
+			$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
+
+			if ( is_array( $output_format ) && array_key_exists( $imagesize['mime'], $output_format ) ) {
```

Comment:
```suggestion
			if ( isset( $output_format[ $imagesize['mime'] ] ) {
```

In my eyes this reduces complexity of the condition and following [isset vs array_key_exists](https://stackoverflow.com/questions/3210935/whats-the-difference-between-isset-and-array-key-exists), this makes no negative difference and would actually be slightly stricter.

Reply:
Thanks for the suggestion. Yes, `isset` would work here too however it would reduce readability a tiny bit. Having the word `array` in the function name makes this bit of code self-documenting. (Also thinking that `array_key_exists` may be using `isset` under the hood, perhaps.)

Reply:
In my eyes this improves the readability, that’s why I suggested it. Since it reduces the complexity.